apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -503,30 +516,44 @@ bump_last_alive(S0) ->
     {ok, replies(), session()}.
 replay(ClientInfo, [], Session0 = #{s := S0}) ->
     Streams = emqx_persistent_session_ds_stream_scheduler:find_replay_streams(S0),
-    Session = lists:foldl(
-        fun({_StreamKey, Stream}, SessionAcc) ->
-            replay_batch(Stream, SessionAcc, ClientInfo)
-        end,
-        Session0,
-        Streams
-    ),
+    Session = replay_streams(Session0#{replay => Streams}, ClientInfo),
```

Comment:
Nit: we could check that `Streams =/= []` before adding the `replay` key; otherwise, there might be a clause error in `handle_timeout` above?

Reply:
Call to `replay_streams/2` should remove the `replay` key (second clause) in that case AFAICS. Anyway, I'll refine guards to make it all look safer.